android/sdk/src/main/java/org/jitsi/meet/sdk/analytics/AmplitudeModule.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright @ 2019-present 8x8, Inc.
+ * Copyright @ 2017-2018 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.meet.sdk.analytics;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+
+import com.amplitude.api.Amplitude;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Implements the react-native module for the Amplitude integration.
+ */
+public class AmplitudeModule
+        extends ReactContextBaseJavaModule {
+
+    public AmplitudeModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    /**
+     * Initializes the Amplitude SDK.
+     *
+     * @param instanceName The name of the Amplitude instance. Should
+     * be used only for multi-project logging.
+     * @param apiKey The API_KEY of the Amplitude project.
+     */
+    @ReactMethod
+    public void init(String instanceName, String apiKey) {
+        Amplitude.getInstance(instanceName).initialize(getCurrentActivity(), apiKey);
+    }
+
+    /**
+     * Sets the user properties for an Amplitude instance.
+     *
+     * @param instanceName The name of the Amplitude instance.
+     * @param userPropsString JSON string with user properties to be set.
+     */
+    @ReactMethod
+    public void setUserProperties(String instanceName, String userPropsString) {
```

Comment:
Why are we getting a JSON encoded string here? Unless it's called multiple times a second, which would incur in a performance penalty, we should be passing an object from JS, which will transform into a ReadableMap.

Reply:
Hmmm. I'm not talking about the JavaScript object passed down to the Java code. I'm talking about the `org.json.JSONObject` parameter of the `setUserProperties` method of the Amplitude Android SDK ( this one https://github.com/amplitude/Amplitude-Android/blob/master/src/com/amplitude/api/AmplitudeClient.java#L1416).

Am I missing something? The link to the documentation that you posted doesn't seem to be related....

Converting `ReadableMap` to **`org.json.JSONObject`** will require going trough all `ReadableMap` properties recursively  and adding them manually to a `org.json.JSONObject`. IMHO this will be uglier! But maybe you have better ideas how to do it! Do you?

Reply:
I think it should be easy: `ReadableMap` has a `toHashMap` function: https://github.com/facebook/react-native/blob/4d95e85f64a496d9b9539b6218d1737997a315da/ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableMap.java#L29 and `JSONObject` accepts a `Map` in its constructor: https://developer.android.com/reference/org/json/JSONObject#JSONObject(java.util.Map)

Can you please give it a try?

Reply:
Sure!